### ISSUE TITLE
Add Bio::Root back

### ISSUE THREAD

**Issue by cjfields** [bioperl/bioperl-live#114]
Consensus (well, handful of :+1: on mail list) want this back in.  Not sure how easily this can be done; @bosborne mentioned he could try taking it on?

**Comment by bosborne** [comment]
Chris,

Yes, I can do this. Can the person who did the dissection explain what they did, so I can resect?

Thanks again,

BIO

**Comment by cjfields** [comment]
The persons would be Rob Buels and myself as part of a GSoC project.  The idea was that Bio::Root would effectively become the true core, but in hindsight that's likely way too granular and seems to cause too many problems for others.  I think a 'clipping the leaves' approach is working better.

Thanks for taking it on.  I think the main changes could probably be pulled back in from the 1.6 branch in bioperl-live, failing that a simple copy (I'm not terribly concerned about loss in history).  I don't think there is much in the separate repo that hasn't also been added to 1.6 tho.

Chris

On Jun 9, 2015, at 8:01 AM, Brian Osborne <notifications@github.com<mailto:notifications@github.com>> wrote:

Chris,

Yes, I can do this. Can the person who did the dissection explain what they did, so I can resect?

Thanks again,

BIO

**Comment by majensen** [comment]
Brian,
You're the man
MAJ

**Comment by fjossandon** [comment]
Hello all!
The last modification to Bio::Root repo was done by me on November last year, and until beginning of January I made sure that the v1.6.x was synched with all the split repos and the main branch, but then I got consumed by job and had to dissappear. =(

Currently the Bio::Root modules from the split repo are almost identical to those on the v1.6.x branch with a few exceptions:
1) v1.6.x branch have a "HTTPget.pm" module that the split branch don't have.
2) v1.6.x branch "Build.pm" module use "=head2" for documentation while the split repo use "=func".
3) v1.6x branch have a "HTTPget.t" test that the split branch don't have and seems like a more developed test file than the split branch extra "xt/RootIO.t" test.
4) v1.6x branch "Exception.t" test use "use lib './examples/root/lib';'" and the split repo use "use lib './eg';" because of their different folder organization.

In summary, I think it would be safe to just copy back the Root from the v1.6.x branch back into master and nothing will be lost.

Cheers,

**Comment by bosborne** [comment]
Francisco,

Excellent, those are the details I needed.

Thanks again,

BIO

**Comment by bioperl-guts** [comment]
@bosborne cool! Let me know if you need anything :+1:

**Comment by bioperl-guts** [comment]
Heh, that was @cjfields replying as @bioperl-guts.  Attempting to debug notifications :smirk:

**Comment by fjossandon** [comment]
Lol, good one!

Looks like is working fine ;)

 

De: bioperl-guts [mailto:notifications@github.com] 
Enviado el: martes, 09 de junio de 2015 13:18
Para: bioperl/bioperl-live
CC: Francisco J. Ossandon
Asunto: Re: [bioperl-live] Add Bio::Root back (#114)

 

Heh, that was @cjfields <https://github.com/cjfields>  replying as @bioperl-guts <https://github.com/bioperl-guts> . Attempting to debug notifications   <https://assets-cdn.github.com/images/icons/emoji/unicode/1f60f.png>

**Comment by bosborne** [comment]
Hmm. I've started working on this, but branched off of master rather than v1.6.x. Was that unwise?

**Comment by majensen** [comment]
I think that's the Right Thing.

On Sat, Jun 20, 2015 at 11:33 AM, Brian Osborne < notifications@github.com [notifications@github.com] > wrote:
Hmm. I've started working on this, but branched off of master rather than
v1.6.x. Was that unwise?



> On Jun 9, 2015, at 12:17 PM, bioperl-guts <notifications@github.com> wrote:
>
> Heh, that was @cjfields <https://github.com/cjfields> replying as
@bioperl-guts <https://github.com/bioperl-guts>. Attempting to debug
notifications
>
> —
> Reply to this email directly or view it on GitHub
<https://github.com/bioperl/bioperl-live/issues/114#issuecomment-110419553>.
>

—
Reply to this email directly or view it on GitHub
[https://github.com/bioperl/bioperl-live/issues/114#issuecomment-113783266] .[https://github.com/notifications/beacon/AAFAmSNaropQpQZrwVeiht0XQoCOlOCcks5oVX9qgaJpZM4E8ME6.gif]

**Comment by bosborne** [comment]
Got it, will proceed ….

**Comment by cjfields** [comment]
That’s correct.  We’re wanting Root back in master, it’s already in 1.6.x :)

-c

On Jun 20, 2015, at 11:17 AM, Brian Osborne <notifications@github.com<mailto:notifications@github.com>> wrote:

Got it, will proceed ….

**Comment by bioperl-guts** [comment]
OK, the branch is merge_root_into_live.

Those t/Tree/Tree.t and t/Tree/TreeIO.t tests continue to be problematic but all other tests pass.

Notes:

- eg/TestObject.pm is now Bio/Root/TestObject.pm
- eg/TestInterface.pm has been removed, looked like it was not used, add it back if you know better
- references to the old Tutorial have been removed
- HTML::TableExtract is recommended, added that to the long list of recommended modules in Build.PL
- a test or 2 needed minor fixes
- a few POD fixes, here and there

Take a look!

**Comment by cjfields** [comment]
No problems on the branch, everything is passing, everything looks good.  On my Mac all `t/Tree/Tree.`t and `t/Tree/TreeIO.t` tests are passing (though I'm running w/o network tests).  So, full steam ahead to merge back in to master!

**Comment by majensen** [comment]
\o/

On Sat, Jun 20, 2015 at 8:07 PM, Chris Fields < notifications@github.com [notifications@github.com] > wrote:
No problems on the branch, everything is passing, everything looks good. On my
Mac all t/Tree/Tree. t and t/Tree/TreeIO.t tests are passing (though I'm running w/o network tests). So, full steam ahead
to merge back in to master!

—
Reply to this email directly or view it on GitHub
[https://github.com/bioperl/bioperl-live/issues/114#issuecomment-113843977] .[https://github.com/notifications/beacon/AAFAmQ5xetq2mXupnQEOv6OhpU7F_ZLDks5oVffJgaJpZM4E8ME6.gif]

**Comment by fjossandon** [comment]
Go go go!!!

 

De: Mark Jensen [mailto:notifications@github.com] 
Enviado el: sábado, 20 de junio de 2015 21:09
Para: bioperl/bioperl-live
CC: Francisco J. Ossandon
Asunto: Re: [bioperl-live] Add Bio::Root back (#114)

 

\o/

On Sat, Jun 20, 2015 at 8:07 PM, Chris Fields < notifications@github.com [notifications@github.com] > wrote:
No problems on the branch, everything is passing, everything looks good. On my
Mac all t/Tree/Tree. t and t/Tree/TreeIO.t tests are passing (though I'm running w/o network tests). So, full steam ahead
to merge back in to master!

—
Reply to this email directly or view it on GitHub
[https://github.com/bioperl/bioperl-live/issues/114#issuecomment-113843977] .[https://github.com/notifications/beacon/AAFAmQ5xetq2mXupnQEOv6OhpU7F_ZLDks5oVffJgaJpZM4E8ME6.gif]

**Comment by bioperl-guts** [comment]
Let’s also talk about install documentation, while we’re at it. We have a bunch of Wiki pages and INSTALL* files saying slightly different things, when they should all say the same thing. Not referring to Windows, that’s something I don’t know much about. IMO these are the essential messages, and should be in this order in the file or page:

- use CPAN
- or if you want to manually install do “git clone …", “perl Build.PL”, etc.
- those who want local installation should use local::lib

We have the text for CPAN already, of course. No mention of “git clone” or local::lib anywhere, as far as I can see.

Overall there’s just way too much text on this topic in the Wiki and in the INSTALL files, should be simpler.

Thoughts?

**Comment by cjfields** [comment]
Re: install docs and so on, see issue #116

**Comment by cjfields** [comment]
Lost track of this one :frowning: .  Merged this into master, so closing ticket.  Documentation issues will be in #116.
